src/MusicStore/Controllers/ShoppingCartController.cs

```diff
@@ -71,24 +71,39 @@ public ShoppingCartController(MusicStoreContext dbContext)
                 .Include(c => c.Album)
                 .SingleOrDefaultAsync();
 
-            // Remove from cart
-            int itemCount = cart.RemoveFromCart(id);
-
-            await DbContext.SaveChangesAsync(requestAborted);
-
-            string removed = (itemCount > 0) ? " 1 copy of " : string.Empty;
-
-            // Display the confirmation message
-
-            var results = new ShoppingCartRemoveViewModel
+            ShoppingCartRemoveViewModel results;
+            if(cartItem != null)
             {
-                Message = removed + cartItem.Album.Title +
-                    " has been removed from your shopping cart.",
-                CartTotal = await cart.GetTotal(),
-                CartCount = await cart.GetCount(),
-                ItemCount = itemCount,
-                DeleteId = id
-            };
+                // Remove from cart
+                int itemCount = cart.RemoveFromCart(id);
+
+                await DbContext.SaveChangesAsync(requestAborted);
+
+                string removed = (itemCount > 0) ? " 1 copy of " : string.Empty;
+
+                // Display the confirmation message
+
+                results = new ShoppingCartRemoveViewModel
```

Comment:
Consider using variables for Message and ItemCount so that the view model creation is not duplicated.